google-cloud-spanner-hibernate-samples/spring-data-jpa-sample/README.adoc

```diff
@@ -0,0 +1,30 @@
+= Cloud Spanner Spring Data JPA with Hibernate Sample
+
+This sample application demonstrates using https://spring.io/projects/spring-data-jpa[Spring Data JPA] with https://hibernate.org/orm/releases/5.4/[Hibernate 5.4] and https://cloud.google.com/spanner/[Google Cloud Spanner].
+
+== Setup & Configuration
+1. Create a Google Cloud Platform Project
+2. https://cloud.google.com/docs/authentication/getting-started#creating_the_service_account[Create a service account] with Cloud Spanner permission.
+Furnish a new JSON key and then set the credentials using the `GOOGLE_APPLICATION_CREDENTIALS` environment variable.
++
+Alternatively, have the https://cloud.google.com/sdk/[Google Cloud SDK] installed and initialized and logged in with https://developers.google.com/identity/protocols/application-default-credentials[application default credentials].
+
+3. Enable the https://console.cloud.google.com/apis/api/spanner.googleapis.com/overview[Cloud Spanner API].
+
+4. In the http://console.cloud.google.com/spanner[Spanner Cloud Console UI], create a Cloud Spanner instance and database and insert those details into the `spring.datasource.url` defined in the file `src/main/resources/application.properties`.
+This setting configures the connection URL which specifies the database that your application will connect to.
++
+----
+# Spanner connection URL.
+# - ${PROJECT_ID} Replace with your GCP project ID
+# - ${INSTANCE_ID} Replace with your Spanner instance ID
+# - ${DATABASE_NAME} Replace with your Spanner database name within your Spanner instance
+
+spring.datasource.url=jdbc:cloudspanner:/projects/${PROJECT_ID}/instances/${INSTANCE_ID}/databases/${DATABASE_NAME}
+----
+
```

Comment:
Maybe explain that the schema will be created on startups and they'll see SQL that Hibernate sends to Spanner in the console.